integration/benchmark_test.go

```diff
@@ -0,0 +1,191 @@
+	_ "github.com/aquasecurity/fanal/analyzer/os/photon"
+	_ "github.com/aquasecurity/fanal/analyzer/os/redhatbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/suse"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/dpkg"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/rpm"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/extractor/docker"
+	"github.com/aquasecurity/fanal/types"
+	dtypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	name      string
+	imageName string
+	imageFile string
+}
+
+var testCases = []testCase{
+	{
+		name:      "happy path, alpine:3.10",
+		imageName: "alpine:3.10",
+		imageFile: "testdata/fixtures/alpine-310.tar.gz",
+	},
+	{
+		name:      "happy path, amazonlinux:2",
+		imageName: "amazonlinux:2",
+		imageFile: "testdata/fixtures/amazon-2.tar.gz",
+	},
+	{
+		name:      "happy path, debian:buster",
+		imageName: "debian:buster",
+		imageFile: "testdata/fixtures/debian-buster.tar.gz",
+	},
+	{
+		name:      "happy path, photon:1.0",
+		imageName: "photon:1.0-20190823",
+		imageFile: "testdata/fixtures/photon-10.tar.gz",
+	},
+	{
+		name:      "happy path, registry.redhat.io/ubi7",
+		imageName: "registry.redhat.io/ubi7",
+		imageFile: "testdata/fixtures/ubi-7.tar.gz",
+	},
+	{
+		name:      "happy path, opensuse leap 15.1",
+		imageName: "opensuse/leap:latest",
+		imageFile: "testdata/fixtures/opensuse-leap-151.tar.gz",
+	},
+	{
+		name:      "happy path, vulnimage with lock files",
+		imageName: "knqyf263/vuln-image:1.2.3",
+		imageFile: "testdata/fixtures/vulnimage.tar.gz",
+	},
+}
+
+func run(ac analyzer.Config, ctx context.Context, tc testCase, b *testing.B) {
+	actualFiles, err := ac.Analyze(ctx, tc.imageFile)
+	require.NoError(b, err)
+
+	osFound, err := analyzer.GetOS(actualFiles)
+	require.NoError(b, err)
+
+	_, err = analyzer.GetPackages(actualFiles)
+	require.NoError(b, err)
+
+	_, err = analyzer.GetPackagesFromCommands(osFound, actualFiles)
+	require.NoError(b, err)
+
+	_, err = analyzer.GetLibraries(actualFiles)
+	require.NoError(b, err)
+}
+
+func runChecksBench(b *testing.B, ac analyzer.Config, ctx context.Context, tc testCase) {
+	for i := 0; i < b.N; i++ {
+		run(ac, ctx, tc, b)
+	}
+}
+
+func BenchmarkFanal_Library_DockerMode_WithoutCache(b *testing.B) {
+	benchCache, _ := ioutil.TempDir("", "BenchmarkFanal_Library_DockerMode_WithoutCache_*")
+	defer os.RemoveAll(benchCache)
+
+	for _, tc := range testCases {
+		ctx, _, _, ac := setup(b, tc, benchCache)
+		b.Run(tc.name, func(b *testing.B) {
+			b.ReportAllocs()
+			b.ResetTimer()
+			runChecksBench(b, ac, ctx, tc)
+			b.StopTimer()
+		})
+
+		// teardown
+		cli, err := client.NewClientWithOpts(client.FromEnv)
+		require.NoError(b, err, tc.name)
```

Comment:
Why don't you initialize this client in `teardown`?

Reply:
good catch. ill fix it.